headers/storage/lock.h

```diff
@@ -0,0 +1,50 @@
+//
+// Created by Sameer Raj on 14/07/24.
+//
+
+#ifndef LOCK_H
+#define LOCK_H
+#include <condition_variable>
+
+
+struct ReadWriteLock {
+    std::mutex mtx;
+    std::atomic_bool isWriterWaiting{false};
+    int readerCount{0};
+    std::atomic_bool isWriteLocked{false};
+    std::condition_variable readCv;
+    std::condition_variable writeCv;
+};
+
+namespace lock {
+    inline void getReadLock(ReadWriteLock& lock) {
```

Comment:
All inline functions in this namespace lock, actually make sense to go as member functions of struct ReadWriteLock